core/commons/che-core-commons-lang/src/test/java/org/eclipse/che/commons/lang/TopologicalSortTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.commons.lang;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.testng.Assert.assertEquals;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class TopologicalSortTest {
+
+  @Test(dataProvider = "sortingCases")
+  public void shouldApplySorting(List<Var> list, List<Var> expectedSorting) throws Exception {
+    // when
+    sort(list);
+
+    // then
+    assertEquals(list, expectedSorting);
+  }
+
+  @DataProvider
+  public static Object[][] sortingCases() {
+    return new Object[][] {
```

Comment:
Here is a valid test case
```java
      // removes duplicates
      new Object[] {
          vars(var("a", "."), var("a", "."), var("b", ".")),
          vars(var("a", "."), var("b", "."))
      }
```
it is expected? if yes - please add note in java doc